### ISSUE TITLE
Support pluggable reporters

### ISSUE THREAD

**Issue by nex3** [dart-lang/unittest#12]
There should be a means for third parties to create reporters that are dynamically loaded and used by unittest.

This is particularly tricky because reporters are likely to be desired on a user-by-user rather than project-by-project basis, so there's some friction to just putting it into a project's pubspec.

**Comment by Fox32** [comment]
I think for the first implementation it would be sufficient if I could specify a reporter via command line. 

The main use case I consider is using the unittest command without a special reporter locally while developing, but another configuration on the continuous integration server. There I might need the report in the JUnit format (like [this](https://github.com/renggli/dart-junit)) or one of the many other formats.

**Comment by nex3** [comment]
I agree; we should definitely also support setting reporters from the command line.

**Comment by nex3** [comment]
This is blocked on #48.

**Comment by jayudey-wf** [comment]
this is needed in order to utilize saucelabs unit testing capabilities, https://docs.saucelabs.com/reference/rest-api/?_ga=1.245459780.1583316598.1430344943#start-js-unit-tests

**Comment by nex3** [comment]
A reporter plugin isn't a good fit for Sauce Labs. A reporter just takes the results of tests being run and displays them to the user, it doesn't run the test itself. #49 or #99 might be a better fit.

**Comment by butlermatt** [comment]
I'd love this functionality, along with #48. I had a previous project using unittest which I wrote my own custom configuration to create Dart Koans (See: https://github.com/butlermatt/dart_koans/blob/master/lib/src/config_koans.dart )

I tried to revive this project this evening with Test but it doesn't seem that same configurability is available with test.
